src/stk/molecular/topology_graphs/host_guest/complex.py

```diff
@@ -133,36 +186,98 @@ def __init__(
-        self._displacement = displacement
+        if guests is not None and guest_start is not None:
+            raise ValueError(
+                'You are attempting to use the old API with the new '
+                'API. Either use the stk.host_guest.Guest API and '
+                '`guests`, or the old API.'
+            )
 
-        super().__init__(
-            building_block_vertices={
+        if guests is not None and guest_target is not None:
+            raise ValueError(
+                'You are attempting to use the old API with the new '
+                'API. Either use the stk.host_guest.Guest API and '
+                '`guests`, or the old API.'
+            )
+
+        if guests is not None:
+            if isinstance(guests, Guest):
+                guests = (guests, )
+
+            building_block_vertices = {
+                host: (HostVertex(0, [0, 0, 0]), )
+            }
+            guest_vertices = {
+                guest.building_block: (GuestVertex(
+                    id=i+1,
+                    position=guest.displacement,
+                    start=guest.start_vector,
+                    target=guest.end_vector,
+                ), )
+                for i, guest in enumerate(guests)
+            }
+            building_block_vertices.update(guest_vertices)
+        elif guest is not None:
+            warnings.warn(
+                'You defined a Complex topology graph using the old '
+                'API (by defining `guest` and any optional arguments: '
+                '`guest_start`, `guest_target` and `displacement`). '
+                'This API will be removed from any version of stk '
+                'released on, or after, 01/01/22. Please use the '
+                '`guests` argument and the `stk.host_guest.Guest` '
+                'dataclass to define all necessary attributes.'
+            )
+            num_nones = sum(
+                1 for vector in (guest_start, guest_target)
+                if vector is None
+            )
+            if num_nones == 1:
+                raise TypeError(
+                    'If guest_start or guest_target is defined, '
+                    'the other must be too.'
+                )
+
+            if guest_start is None:
+                start = target = (1., 0., 0.)
+            else:
+                start = guest_start = tuple(guest_start)
+                target = guest_target = tuple(guest_target)
+
+            if displacement is None:
+                displacement = (0, 0, 0)
+
+            # Save the values as None, for __repr__.
+            self._guest_start = guest_start
+            self._guest_target = guest_target
+            self._displacement = displacement
+            building_block_vertices = {
                 host: (HostVertex(0, [0, 0, 0]), ),
                 guest: (
                     GuestVertex(1, displacement, start, target),
                 ),
-            },
+            }
```

Comment:
There should probaly be an `else` statement here (ie if the user did not define either the `guests` or the `guest`, which raises a runtime error and tells the user they need to use the `guests` parameter